src/App/SentryErrorInfo/index.tsx

```diff
@@ -0,0 +1,23 @@
+import { setTag, setUser } from '@sentry/react';
+import React, { useEffect } from 'react';
+
+import { useAuth } from 'context/AuthContext';
+
+export const SentryErrorInfo: React.FC = () => {
+  const { accountAddress, chainId } = useAuth();
+
+  // Enriches Sentry events with info about the user and chain
+  useEffect(() => {
+    if (accountAddress) {
+      setUser({
+        id: accountAddress,
+      });
+    } else {
+      setUser(null);
+    }
+
+    setTag('chainId', chainId);
+  }, [accountAddress, chainId]);
```

Comment:
I'd separate this effect into two: one for the account address, one for the chain ID. Otherwise both the user and the chainId tag will be updated whenever one of them changes.